collections/sum_of_test.ts

```diff
@@ -0,0 +1,88 @@
+// Copyright 2018-2021 the Deno authors. All rights reserved. MIT license.
+
+import { assertEquals } from "../testing/asserts.ts";
+import { sumOf } from "./sum_of.ts";
+
+Deno.test("[collections/sumOf] On object properties", () => {
+  const object = [
+    { name: "Kyle", age: 34 },
+    { name: "John", age: 42 },
+    { name: "Anna", age: 23 },
+  ];
+
```

Comment:
I would add test for

- the function not mutating the original array that was passed (see all other function test suites in the module)
- empty input array should result in a sum of 0
- `NaN` and `Infinity` behaviour